src/auth/ejabberd_auth_dummy.erl

```diff
@@ -52,50 +48,40 @@ check_password(_HostType, User, Server, _Password, _Digest, _DigestGen) ->
                  user => User, server => Server}),
     false.
 
-%% @spec (User::string(), Server::string(), Password::string()) ->
-%%       ok | {error, invalid_jid}
+-spec set_password(mongooseim:host_type(), jid:luser(), jid:lserver(), binary()) -> ok.
 set_password(_HostType, _User, _Server, _Password) ->
     ok.
 
-%% @spec (User, Server, Password) -> {atomic, ok} | {atomic, exists} | {error, invalid_jid} | {aborted, Reason}
+-spec try_register(mongooseim:host_type(), jid:luser(), jid:lserver(), binary()) -> ok.
 try_register(_HostType, _User, _Server, _Password) ->
     ok.
 
-dirty_get_registered_users() ->
-    [].
-
-get_vh_registered_users(_Server) ->
-    [].
-
-get_vh_registered_users(_Server, _) ->
-    [].
-
-get_vh_registered_users_number(_Server) ->
-    0.
-
-get_vh_registered_users_number(_Server, _) ->
-    0.
-
-get_password(_User, _Server) ->
+-spec get_password(mongooseim:host_type(), jid:luser(), jid:lserver()) -> binary().
+get_password(_HostType, _User, _Server) ->
     <<>>.
 
-get_password_s(_User, _Server) ->
+-spec get_password_s(mongooseim:host_type(), jid:luser(), jid:lserver()) -> binary().
+get_password_s(_HostType, _User, _Server) ->
     <<>>.
 
-%% @spec (User, Server) -> true | false | {error, Error}
-does_user_exist(_User, _Server) ->
+-spec does_user_exist(mongooseim:host_type(), jid:luser(), jid:lserver()) -> boolean().
+does_user_exist(_HostType, _User, _Server) ->
     true.
 
 %% @spec (User, Server) -> ok
 %% @doc Remove user.
 %% Note: it returns ok even if there was some problem removing the user.
-remove_user(_User, _Server) ->
+-spec remove_user(mongooseim:host_type(), jid:luser(), jid:lserver()) -> ok.
+remove_user(_HostType, _User, _Server) ->
     ok.
 
+-spec remove_domain(mongooseim:host_type(), jid:lserver()) -> ok.
```

Comment:
I think this is not entirely correct typing - there is a `simple_acc` type in `mongoose_hooks`, which is just a map, and it's used only for the remove domain hooks. I would personally propose getting rid of this type from `mongoose_hooks`, but it may serve some purpose. It's hard to tell, because the hook doesn't seem to be run from any code apart from the tests.
The other comment is that this handler is present only in the dummy module out of the auth modules, right? Is it an oversight?

Reply:
Oh, sorry, for some reason I misread the spec... Obviously this is fine - I'll edit the comment. But still it's a callback that is present only in the dummy module - is it deliberate?

Reply:
Handling domain removal has to be added to the TODO list. Doing it now.